### DISCUSSION TITLE
Bigger width for the web interface

### DISCUSSION THREAD

**Discussion by De-M-oN** [rejetto/hfs#122] [Ideas]
I really like the new web interface, but can you make it a bit wider (maybe even screen-aware width?)

Because its really thin and I have just 16:10 aspect ratio. On 16:9 it would be even thinner.

This is how it looks on 2560x1600

https://abload.de/img/screenshot2023-02-2804ifgu.png

Its quite thin

**Comment by rejetto** [comment]
Can you show me what you have in mind?
Because a.wider list would be harder to read, a lot of space between name and size.

**Comment by De-M-oN** [comment]
hmm then maybe 50% wider? This seems a bit very thin to me.

Maybe +50% could work

**Comment by rejetto** [comment]
this is now
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/1367199/221843181-d1eea5d9-47dd-43ce-b47c-babe02349b97.png">

this is +50% you suggest
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/1367199/221843283-0189d332-7784-42d1-8f7d-af481ff95686.png">

nicer to look at, but harder to read.
I'm ok to make it nicer while it doesn't compromise reading.

Anyway, you can do whatever you want with your server, and with little html knowledge you can customize the look of it.
For example you can:
admin > config > custom header

    <style>#root { max-width: 75em; }</style>

<img width="495" alt="image" src="https://user-images.githubusercontent.com/1367199/221844214-3089c8c2-189c-48df-a2e0-c88c327293d4.png">

**Reply by rejetto** [reply]
p.s. you can directly paste images here, while you type. Very convenient 😉

**Comment by De-M-oN** [comment]
+50% is better
especially If you have a longer filename its better when it doesnt happen to be a 2 lines object. More width prevents that.
Thank you for the custom header code. Changing it for myself is fine.

**Comment by rejetto** [comment]
next release will replace custom-header feature with another one, more powerful.
your old option should automatically be imported, tho.
i'm telling you just in case something goes wrong 😄

**Comment by De-M-oN** [comment]
It seems the resolution being the issue. Because on your end it is already much wider. You're probably using 1080p?

I have 2560x1600 and this is even a 16:10 aspect ratio and still it is much thinner on my end.

Look how the default looks for me:

![Screenshot 2023-02-28 170618](https://user-images.githubusercontent.com/18665730/221910495-69a6a8c9-adc6-445f-b206-44b01bb78976.png)

**Comment by rejetto** [comment]
i have a 4k screen that is handled like a 1920 by macOs, so yes, that accounts for the difference between our screenshots.
Anyway, my considerations weren't related to the "ugly" blank sides.
I consider current width to be a good middle measure to fit short and middle file names, that are in my experience the majority.
I'm sure some people will serve mostly files with long names and will not be satisfied with that, but until I have a solution to fit both worlds, there's no point in fixing one while ruining the other.

Next HFS version will make possible to make a plugin that will easily provide few customizations, like this, in a easier form (easier then the <style> thing, i mean).
I may even make this plugin myself.

**Reply by ZIKVLD** [reply]
Are you not yet ready to write this plugin?
PS 35" 4k is hard to work with.
And put the settings here:
![Снимок экрана 2024-02-10 172026](https://github.com/rejetto/hfs/assets/154645533/76ab0e6a-a2c9-4617-bc90-a43ecc8b2974)

**Reply by rejetto** [reply]
hey, before talking about a plugin, did you try to change the size "manually" as it is described above?
please test a different size, be sure that you are happy with it, and please report here what the good size is for you.

**Reply by ZIKVLD** [reply]
I don't have a custom header option in Custom HTML

**Reply by ZIKVLD** [reply]
<style>#root { max-width: 75em; }</style>
Does not work. There is a shift to the left

**Reply by rejetto** [reply]
yeah, sorry, that has changed, you can use section "top of html body" (in custom html)

**Reply by rejetto** [reply]
<style>
.list-wrapper { max-width: 100em; }
</style>

**Reply by rejetto** [reply]
i added this example in documentation
https://github.com/rejetto/hfs/wiki/Customization#set-a-wider-list

**Reply by rejetto** [reply]
#root was good for older versions, now it has changed

**Reply by ZIKVLD** [reply]
I saw this option in the code
.list-wrapper { max-width: 54em; }
But the description above confused me.
Now everything has fallen into place.
Thank you.

**Reply by rejetto** [reply]
good, but 54em is the default size.
I'd like to know what is the size you like for your screen.

**Reply by ZIKVLD** [reply]
For me
.list-wrapper { max-width: 75em; }
just right

**Reply by rejetto** [reply]
ok, thanks for the feedback

**Comment by De-M-oN** [comment]
My point was, that what you see is not what I see on a bigger resolution.
So for me it is even thinner than on your end with 1080p size.
Imagine now unscaled 4k.
It doesnt scale with the resolution which makes it thinner as bigger resolution as you have. But an advantage of bigger resolution is on the other hand of course the more available space on the vertical

resolution-aware width (scales the width with resolution) and with custom size settable would be probably the best

**Comment by rejetto** [comment]
It's not that I'm lazy 😄, the width would automatically take advantage of the whole screen, that's the default behaviour.
I, on purpose, decided to restrict the width, by introducing a "style rule", because I think it's good, for the reason i explained earlier.

While on your screen the effect is "stronger", you can see that on my screen I too have a lot of wasted screen space.
